lib/galaxy/webapps/galaxy/services/datasets.py

```diff
@@ -981,3 +981,12 @@ def _raw_data(
         data = data_provider.get_data(**kwargs)
 
         return data
+
+    def _get_indexer(self, trans, dataset):
+        try:
+            indexer = self.data_provider_registry.get_data_provider(trans, original_dataset=dataset, source="index")
+            if indexer is None:
+                raise Exception("No indexer available for this dataset")
```

Comment:
You probably can directly raise the galaxy_exception here and remove the "try/catch all".

Also, suppose you expect other unhandled exceptions to happen when calling get_data_provider. In that case, it is probably better to `log` that exception instead of raising a `galaxy_exception`, as those can "leak" internal details to the user. But sometimes it could be helpful, depending on the possible errors you expect.

Reply:
Just wondering, if we log instead of raising an exception, that will still return an `Internal Server Error` (like it does rn without these changes) from the api instead of an exception/err_msg? In that case, this change wouldn't really be needed then, right?

Reply:
That is likely fine, they can be caught somewhere else or we just did not know about the possibility. Masking them here with a wide `except Exception` might create a problem for future us.

Reply:
Sorry, I was not very clear in my comment. I was talking about 2 scenarios:

- **Removing the try/catch-all** and explicitly raising a `galaxy_exception` when the indexer is None. This will still raise an `Internal Server Error` in other possible exception cases (which is fine). You don't need to log anything.

- **Keep the try/catch-all** and still raise the galaxy_exception as in the previous case. Here, in the catch, you can log the error and `re-raise` the exception. This will result in an `Object not found` error when the indexer is None and an `Internal Server Error` when something else goes wrong (the same as the previous case). The important part is *not to raise a `galaxy_exception` in the catch-all*, as this will result in any internal "unexpected" error leaking into the response.

Reply:
Ah I see, that makes sense, thank you for explaining! I ended up removing the try/catch-all and just raised the exception. Now, from what you explained, an exception will explicitly be raised for the case indexer=None, and nothing else, not exposing any potential variables from the api.